core/services/relay/evm/chain_reader.go

```diff
@@ -456,6 +445,52 @@ func getEventTypes(event abi.Event) ([]abi.Argument, types.CodecEntry, map[strin
 	return indexedAsUnIndexedTypes, types.NewCodecEntry(indexedTypes, nil, nil), dataWords
 }
 
+func findFieldIndex(arg abi.Argument, fieldPath string, dataWords map[string]read.DataWordDetail, index int) int {
```

Comment:
wdyt about and add a short comment
```suggestion
func calculateFieldDWIndex(arg abi.Argument, fieldPath string, dataWords map[string]read.DataWordDetail, index int) int {
```

Reply:
refactored field name and added a comment above